### PR TITLE
Update tpms_eezrv.c

### PR DESCRIPTION
**Pull request by flyboy013**
Checksum and pressure bug fixes.  Add leak detection.

### CODE REVIEW

File: src/devices/tpms_eezrv.c [hunk 1]
```diff
@@ -14,28 +14,28 @@
...
-    PRE CC IIIIII PP TT FF 00
+    PRE CC IIIIII PP TT FF FF
 
 - PRE : FFFF
 - C : 8 bit CheckSum
 - I: 24 bit ID
 - P: 8 bit pressure  P * 2.5 = Pressure kPa
 - T: 8 bit temperature   T - 50 = Temperature C
-- F: 8 bit battery (not verified) and deflation pressure (for sure) flags
+- F: 16  bit battery (not verified), deflation pressure (for sure), pressure modifier flags
```
Comment (zuckschwerdt) [inline src/devices/tpms_eezrv.c:30]: `640/2.5 = 256` ;) so please: "pressure MSB"

File: src/devices/tpms_eezrv.c [hunk 2]
```diff
@@ -55,36 +55,50 @@ static int tpms_eezrv_decode(r_device *decoder, bitbuffer_t *bitbuffer)
...
+    uint8_t b[7]  = {0};
     uint8_t cc[1] = {0};
     bitbuffer_extract_bytes(bitbuffer, 0, pos + 16, cc, sizeof(cc) * 8);
     bitbuffer_extract_bytes(bitbuffer, 0, pos + 24, b, sizeof(b) * 8);
+
     // verify checksum
-    if ((add_bytes(b, 6) & 0xff) != cc[0]) {
+    int computed_checksum = add_bytes(b, sizeof(b));
+    if (computed_checksum > 0xff) 
+        computed_checksum |= 0x80;
```
Comment (zuckschwerdt) [inline src/devices/tpms_eezrv.c:73]: Can you add a comment that the top bits fold around? That's what you mean here, right?
Did you observe a checksum bigger than 1FF ? I.e. what would 2FF do?

File: src/devices/tpms_eezrv.c [hunk 3]
```diff
@@ -55,36 +55,50 @@ static int tpms_eezrv_decode(r_device *decoder, bitbuffer_t *bitbuffer)
...
         return DECODE_FAIL_MIC;
     }
     char id_str[7];
     sprintf(id_str, "%02x%02x%02x", b[0], b[1], b[2]);
     float pressure_kPa = b[3] * 2.5;
     int temperature_C  = b[4] - 50;
-    int flags          = b[5];
-    char flags_str[3];
-    sprintf(flags_str, "%x", flags);
+    int flags1          = b[5];
```
Comment (zuckschwerdt) [inline src/devices/tpms_eezrv.c:83]: are these the same indent as above? looks like one extra space?

File: src/devices/tpms_eezrv.c [hunk 4]
```diff
@@ -55,36 +55,50 @@ static int tpms_eezrv_decode(r_device *decoder, bitbuffer_t *bitbuffer)
...
+    int flags2          = b[6];
+    int fast_leak       = 0;
+
+    fast_leak = (flags1 & 0x10) && !(flags1 & 0x20);   // Check for fast leak flag set and resolved flag NOT set
```
Comment (zuckschwerdt) [inline src/devices/tpms_eezrv.c:87]: perhaps write as
```
int leak_flag = (flags1 & 0x10);
int resolved_flag = (flags1 & 0x20);
int fast_leak = leak_flag && !resolved_flag;
```

**Comment by zuckschwerdt** [comment]
Rebase if possible (otherwise I can do that). Some minor comments, otherwise looks good!

File: src/devices/tpms_eezrv.c (hunk 2 above)
Comment (flyboy013) [reply src/devices/tpms_eezrv.c:73]: If the checksum is greater than 0xFF then the MSB is set.  I occurs whether the bit is already set or not and was observed when checksum was in the 0x1FF and the 0x2FF range.  I did not observe a checksum higher than in the 0x200-0x2FF range.  I'm assuming this holds true for higher than 0x2FF, but I have not data to support it.

**Comment by zuckschwerdt** [comment]
Did you overwrite your changes? Don't panic, git still has all the data. Read up on git reflog, e.g. https://stackoverflow.com/questions/10099258/how-can-i-recover-a-lost-commit-in-git

File: src/devices/tpms_eezrv.c (hunk 1 above)
Comment (flyboy013) [reply src/devices/tpms_eezrv.c]: Changes made to add value in flag field (currently on 1 bit) to the pressure as the MSB.

File: src/devices/tpms_eezrv.c (hunk 4 above)
Comment (flyboy013) [reply src/devices/tpms_eezrv.c]: Code updated to note value of fast_leak_detected and fast_leak_resolved and to then determine reported fast_leak output field base on these.

**Comment by flyboy013** [comment]
PR accidentally deleted and resubmitted as PR  #2664

**Comment by zuckschwerdt** [comment]
Note for posteriority: we prefer that branches be kept (force push into the same PR) and not resubmitted.

**Pull request closed without merging**